pkg/connect/state/state.go

```diff
@@ -268,15 +274,18 @@ func (g *WorkerGroup) Sync(ctx context.Context, groupManager WorkerGroupManager,
 	}
 
 	// Update the worker group to make sure it store the appropriate IDs
-	if syncReply.IsSuccess() {
-		g.AppID = syncReply.AppID
-		g.SyncID = syncReply.SyncID
-
-		// Update the worker group with the syncID so it's aware that it's already sync'd before
-		// Always update the worker group for consistency, even if the context is cancelled
-		if err := groupManager.UpdateWorkerGroup(context.Background(), g.EnvID, g); err != nil {
-			return fmt.Errorf("error updating worker group: %w", err)
-		}
+	if !syncReply.IsSuccess() {
```

Comment:
Previously, we ignored invalid sync responses. That's not good. We should always prevent connections from being established in case of a connection requiring a successful sync and not receiving a successful response.